core/variable.h

```diff
@@ -115,6 +166,184 @@ template <class T> class deep_ptr {
   std::unique_ptr<T> m_data;
 };
 
+namespace detail {
+template <class T>
+std::unique_ptr<VariableConceptT<T>>
+makeVariableConceptT(const Dimensions &dims);
+template <class T>
+std::unique_ptr<VariableConceptT<T>>
+makeVariableConceptT(const Dimensions &dims, Vector<T> data);
+} // namespace detail
+
+template <class Op> struct TransformInPlace {
+  Op op;
+  template <class T> void operator()(T &&handle) const {
+    auto data = handle->getSpan();
+    std::transform(data.begin(), data.end(), data.begin(), op);
+  }
+  template <class A, class B> void operator()(A &&a, B &&b_ptr) const {
+    // std::unique_ptr::operator*() is const but returns mutable reference, need
+    // to artificially put const to we call the corect overloads of ViewModel.
+    const auto &b = *b_ptr;
```

Comment:
In practice this should not happend for valid variables. However there is https://github.com/scipp/scipp/issues/206. What you comment on is probably one of several issues, but likely it will actually fail (segfault) before reaching this point in code, e.g., when doing a dimension check.